mne/io/base.py

```diff
@@ -1086,17 +1087,47 @@ def apply_function(
         if dtype is not None and dtype != self._data.dtype:
             self._data = self._data.astype(dtype)
 
+        args = getfullargspec(fun)[0]
+        if "ch_idx" in args:
+            logger.info("apply_function requested to access ch_idx")
+        elif "ch_name" in args:
+            logger.info("apply_function requested to access ch_name")
+
```

Comment:
I think a warning or error should be thrown if both parameters are found, instead of silently ignoring ch_name

Reply:
Rather than `if`/`elif` I don't see a problem with using `if/if`, i.e., providing both if they are both in the signature

Reply:
@hoechenberger i had already implemented this, but not pushed yet. done by now
@larsoner: i had discussed this with @drammock and @sappelhoff, and they argued for excluding this option to avoid potential trouble.

Reply:
Okay!